src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/utils/csv/CaseReferenceCsvLoader.java

```diff
@@ -31,4 +47,61 @@ public List<CaseReference> loadCaseReferenceList(String fileName) {
         }
     }
 
+    public List<CaseReference> loadCaseReferenceList(String fileName, String secret) {
+        try {
+            String encryptedContent = new String(Files.readAllBytes(Paths.get(getClass().getClassLoader().getResource(fileName).toURI())));
+            SecretKey key = getKeyFromString(secret);
+            String decryptedContent = decrypt(encryptedContent, key);
+
+            CsvMapper csvMapper = new CsvMapper();
+            CsvSchema csvSchema = csvMapper.typedSchemaFor(CaseReference.class).withHeader();
+            List list = new CsvMapper().readerFor(CaseReference.class)
+                    .with(csvSchema)
+                    .readValues(decryptedContent)
+                    .readAll();
+            return list;
+        } catch (Exception e) {
+            log.error("Error occurred while loading object list from file " + fileName, e);
+            return Collections.emptyList();
+        }
+    }
+
+    public static SecretKey getKeyFromString(String key) throws Exception {
+        SecretKeySpec secretKey = generateKey(key);
+        return secretKey;
+    }
+
```

Comment:
The migration itself is more important, but make a note to update some documentation how to encrypt using these functions.  Readme or Confluence, whichever is more appropriate.  Be useful for the next dev using this :-)